cmd/state-custom-installer/runInstall.go

```diff
@@ -0,0 +1,188 @@
+	"github.com/ActiveState/cli/internal/runbits/buildlogfile"
+	"github.com/ActiveState/cli/internal/subshell"
+	"github.com/ActiveState/cli/internal/subshell/sscommon"
+	"github.com/ActiveState/cli/internal/testhelpers/outputhelper"
+	"github.com/ActiveState/cli/internal/unarchiver"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/ActiveState/cli/pkg/platform/runtime/setup/events"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+func runInstall(out output.Outputer, params *Params) error {
+
+	tempDir, err := ioutil.TempDir("", "artifacts-")
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// #FIX: Need to uncomment this for production
+	defer os.RemoveAll(tempDir)
+
+	fmt.Printf("Temp directory is: %s\n", tempDir)
+
+	installToDir := params.path
+	artifactsTarGz := "artifacts.tar.gz"
+	assetsPath := filepath.Join(tempDir, "assets")
+	artifactsPath := filepath.Join(tempDir, "artifacts")
+	licenseFilePath := filepath.Join(assetsPath, "LICENSE.txt")
+
+	// Double check if installation directory exists already
+	if fileutils.DirExists(installToDir) {
+		empty, err := fileutils.IsEmptyDir(installToDir)
+		if err == nil && !empty {
+			configureEnvironmentAccepted, err := getConfirmation("Installation directory is not empty, install anyway?")
+			if err != nil {
+				return err
+			}
+
+			if !configureEnvironmentAccepted {
+				return errs.New("Installation directory not empty, installation aborted")
+			}
+		}
+	}
+
+	if err := os.Mkdir(assetsPath, os.ModePerm); err != nil {
+		return err
+	}
+
+	if err := os.Mkdir(artifactsPath, os.ModePerm); err != nil {
+		return err
+	}
+
+	err = unzip(os.Args[0], assetsPath)
+	if err != nil {
+		return err
+	}
+
+	eulaAccepted, err := showEULAAndGetAcceptance(licenseFilePath)
+	if err != nil {
+		return err
+	}
+
+	if !(eulaAccepted) {
+		return errors.New("\n\nEnd user didn't accept the EULA")
+	}
+
+	fmt.Println("Stage 1 of 2 Start: Decompressing assets into: ", assetsPath)
+	err = unzip(os.Args[0], assetsPath)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Stage 1 of 2 Finished: Decompressing assets into: ", assetsPath)
+
+	archivePath := filepath.Join(assetsPath, artifactsTarGz)
+
+	fmt.Println("Stage 2 of 2 Start: Decompressing artifacts into: ", artifactsPath)
+	err = unpackArtifact(unarchiver.NewTarGz(), archivePath, artifactsPath)
+	if err != nil {
+		err := errs.Wrap(err, "Could not unpack artifact %s", archivePath)
+		return err
+	}
+	fmt.Println("Stage 2 of 2 Finished: Decompressing artifacts into: ", artifactsPath)
+
+	owner := "voidOwner"
+	name := "voidProject"
+	commitID := strfmt.UUID("000000000-0000-0000-0000-000000000000")
+
+	offlineTarget := target.NewOfflineTarget(owner, name, commitID, installToDir, artifactsPath)
+
+	analytics := blackhole.New()
+	mockProgress := &OfflineProgressOutput{}
+	logfile, _ := buildlogfile.New(outputhelper.NewCatcher())
```

Comment:
Please don't ever leave errors unhandled. Go does not have exceptions, errors are your lifeline.